src/vellum/workflows/nodes/displayable/tool_calling_node/tests/test_composio_node.py

```diff
@@ -0,0 +1,123 @@
+import pytest
+from contextlib import contextmanager
+import json
+from unittest.mock import Mock, patch
+
+from vellum.client.types.function_call import FunctionCall
+from vellum.client.types.function_call_vellum_value import FunctionCallVellumValue
+from vellum.workflows.errors.types import WorkflowErrorCode
+from vellum.workflows.exceptions import NodeException
+from vellum.workflows.nodes.displayable.tool_calling_node.state import ToolCallingState
+from vellum.workflows.nodes.displayable.tool_calling_node.utils import ComposioNode
+from vellum.workflows.types.definition import ComposioToolDefinition
+
+
+@pytest.fixture
+def composio_tool():
+    """Fixture for a sample ComposioToolDefinition."""
+    return ComposioToolDefinition(
+        toolkit="GITHUB", action="GITHUB_CREATE_AN_ISSUE", description="Create a new issue in a GitHub repository"
+    )
+
+
+@pytest.fixture
+def mock_state():
+    """Fixture for a mock ToolCallingState."""
+    state = Mock(spec=ToolCallingState)
+    state.chat_history = Mock()
+    return state
```

Comment:
[nit] In general, we try to only mock external deps (eg api requests), and reuse our source code where possible.

so alternatives available, in order of increasing preference:
- Our test just instantiates `ComposioNode(state=ToolCallingState)` in the test.
- we test `create_function_node` instead, which captures what we're mocking
- we test a workflow that uses a ToolCallingNode with a Composio tool definition just as a user would define it